### PR TITLE
Fix issue with 32bit architecture on big numbers' random generation

### PR DESCRIPTION
**Pull request by iltoga**
Fix issue with 32bit architecture on big numbers' random generation

### CODE REVIEW

File: p2p/strategy/priorityStrategy.go [hunk 1]
```diff
@@ -1106,10 +1106,10 @@ func (ps *PriorityStrategy) GetAnyUnresolvedPeer() *model.Peer {
 	if len(unresolvedPeers) < 1 {
 		return nil
 	}
-	randomIdx := int(util.GetSecurePositiveRandom()) % len(unresolvedPeers)
+	randomIdx := uint64(util.GetSecurePositiveRandom()) % uint64(len(unresolvedPeers))
 	idx := 0
 	for _, peer := range unresolvedPeers {
-		if idx == randomIdx {
+		if uint64(idx) == randomIdx {
```
Comment (capt4ce) [inline p2p/strategy/priorityStrategy.go:1112]: I think this casting should better be performed when we initiate `idx := 0` right?

File: p2p/strategy/priorityStrategy.go [hunk 2]
```diff
@@ -1310,10 +1310,10 @@ func (ps *PriorityStrategy) GetAnyKnownPeer() *model.Peer {
 	if len(knownPeers) < 1 {
 		panic("No well known peer is found")
 	}
-	randomIdx := int(util.GetSecurePositiveRandom()) % len(knownPeers)
+	randomIdx := uint64(util.GetSecurePositiveRandom()) % uint64(len(knownPeers))
 	idx := 0
 	for _, peer := range knownPeers {
-		if idx == randomIdx {
+		if uint64(idx) == randomIdx {
```
Comment (capt4ce) [inline p2p/strategy/priorityStrategy.go:1316]: I think this casting should better be performed when we initiate idx := 0 right?

File: p2p/strategy/peerStrategyHelper.go [hunk 3]
```diff
@@ -77,10 +77,10 @@ func (ps *PeerStrategyHelper) GetRandomPeerWithoutRepetition(peers map[string]*m
 	var (
 		peer *model.Peer
 	)
-	randomIdx := int(util.GetSecurePositiveRandom()) % len(peers)
+	randomIdx := uint64(util.GetSecurePositiveRandom()) % uint64(len(peers))
 	idx := 0
 	for _, knownPeer := range peers {
-		if idx == randomIdx {
+		if uint64(idx) == randomIdx {
```
Comment (capt4ce) [inline p2p/strategy/peerStrategyHelper.go:83]: I  think this casting should better be performed when we initiate idx := 0 right?

File: p2p/strategy/nativeStrategy.go [hunk 4]
```diff
@@ -493,10 +493,10 @@ func (ns *NativeStrategy) GetAnyKnownPeer() *model.Peer {
 	if len(knownPeers) < 1 {
 		panic("No well known peer is found")
 	}
-	randomIdx := int(util.GetSecurePositiveRandom()) % len(knownPeers)
+	randomIdx := uint64(util.GetSecurePositiveRandom()) % uint64(len(knownPeers))
 	idx := 0
 	for _, peer := range knownPeers {
-		if idx == randomIdx {
+		if uint64(idx) == randomIdx {
```
Comment (capt4ce) [inline p2p/strategy/nativeStrategy.go:499]: I think this casting should better be performed when we initiate idx := 0 right?

File: p2p/strategy/nativeStrategy.go [hunk 5]
```diff
@@ -365,10 +365,10 @@ func (ns *NativeStrategy) GetAnyUnresolvedPeer() *model.Peer {
 	if len(unresolvedPeers) < 1 {
 		return nil
 	}
-	randomIdx := int(util.GetSecurePositiveRandom()) % len(unresolvedPeers)
+	randomIdx := uint64(util.GetSecurePositiveRandom()) % uint64(len(unresolvedPeers))
 	idx := 0
 	for _, peer := range unresolvedPeers {
-		if idx == randomIdx {
+		if uint64(idx) == randomIdx {
```
Comment (capt4ce) [inline p2p/strategy/nativeStrategy.go:371]: I think this casting should better be performed when we initiate idx := 0 right?

File: p2p/strategy/nativeStrategy.go [hunk 6]
```diff
@@ -305,13 +305,13 @@ func (ns *NativeStrategy) GetAnyResolvedPeer() *model.Peer {
...
-	randomIdx := int(util.GetSecurePositiveRandom())
+	randomIdx := uint64(util.GetSecurePositiveRandom())
 	if randomIdx != 0 {
-		randomIdx %= len(resolvedPeers)
+		randomIdx %= uint64(len(resolvedPeers))
 	}
 	idx := 0
 	for _, peer := range resolvedPeers {
-		if idx == randomIdx {
+		if uint64(idx) == randomIdx {
```
Comment (capt4ce) [inline p2p/strategy/nativeStrategy.go:314]: I think this casting should better be performed when we initiate idx := 0 right?

File: p2p/strategy/priorityStrategy.go (hunk 1 above)
Comment (iltoga) [reply p2p/strategy/priorityStrategy.go]: ok...

File: p2p/strategy/priorityStrategy.go (hunk 2 above)
Comment (iltoga) [reply p2p/strategy/priorityStrategy.go]: ok...

File: p2p/strategy/peerStrategyHelper.go (hunk 3 above)
Comment (iltoga) [reply p2p/strategy/peerStrategyHelper.go]: ok...

File: p2p/strategy/nativeStrategy.go (hunk 4 above)
Comment (iltoga) [reply p2p/strategy/nativeStrategy.go]: ok...

File: p2p/strategy/nativeStrategy.go (hunk 5 above)
Comment (iltoga) [reply p2p/strategy/nativeStrategy.go]: ok..

File: p2p/strategy/nativeStrategy.go (hunk 6 above)
Comment (iltoga) [reply p2p/strategy/nativeStrategy.go]: ok...

**Review by capt4ce — APPROVED** [review]

**Pull request merged**